src/cdi_ndi/NDI_Base.hh

```diff
@@ -122,6 +111,11 @@ protected:
   NDI_Base(const NDI_Base &) = delete;
 
 public:
+  //! Use a gendir.all file other than ${NDI_ROOT_DIR}/share/gendir.all
+  void override_gendir_path(const std::string new_gendir_path) {
+    gendir = new_gendir_path;
```

Comment:
Add a DbC for `fileExists`.

Reply:
`override_gendir_path` is now removed in favor of overloading the `NDI_Base` constructor.